wasm-text.md

```diff
@@ -411,9 +405,10 @@ Since we do not have polymorphic functions available, we define one function per
     syntax TypeUse   ::= "#t2aTypeUse"    "<" Context ">" "(" TypeUse    ")" [function]
     syntax LocalDecl ::= "#t2aLocalDecl"  "<" Context ">" "(" LocalDecl  ")" [function]
  // -----------------------------------------------------------------------------------
-    rule text2abstract(SS) => #t2aStmts<ctx( ... localIds: .Map)>(SS)
+    rule text2abstract(DS:Defns) => text2abstract(( module DS ) .Stmts)
+    rule text2abstract(SS)       => #t2aStmts<ctx( ... localIds: .Map)>(SS) [owise]
 
-    rule #t2aStmt<C>(( module OID:OptionalId DS )) => ( module OID #t2aDefns<C>(DS) )
+    rule #t2aStmt<C>(( module OID:OptionalId DS )) => ( module OID #t2aDefns<C>(unfoldDefns(DS)) )
```

Comment:
I'm not sure it makes sense to entangle these two functions does it? Is it possible to do the `unfold` pass completely then the `#t2a` pass? Then `#text2abstract(D) => #t2a<...>(unfoldDefns(...)` or something like that?

Reply:
Not sure it makes sense, just thinking out loud here.

Reply:
We could do an initial pass that only does unfolding. So we could do

```
unfoldStmts(( module OID:OptionalId DS) SS) => (module OID unfoldDefns(DS)) unfoldStmts(SS)
```

Then maybe a module structuring pass:

```
structureModules((module OID:OptionalId DS) SS) => sortModule((module OID DS)) structureModules(SS)
```

And then finally the existing `#t2a` pass (we can have a better name maybe, now that it is more specialized.

```
#t2aStmts<_>(#module(... funcs: FS, tables: TS, .......) #as M)
=> #t2aModule<#freshModuleContext(FS, TS, ...)>(M)
```
where `#freshModuleContext` produces a context with all the identifers extracted.

Then we get
```
text2abstract(DS:Defns) => text2abstract((module DS) .Stmts)
text2abstract(SS) => #t2aStmts(structureModules((unfoldStmts(SS))) [owise]
```

Reply:
`#t2aModule` would just distrubte the `#t2aDefns` with the new context out over all the different defintion types in the record.

Reply:
Done